File: CpuThrottlingTest/PendingResultStore.swift
import Foundation

struct PendingTestResult: Codable, Identifiable {
    let id: UUID
    let timestamp: Double
    let durationSeconds: Int
    let testDurationType: Int // 0 = 5 min, 1 = 15 min, 2 = 30 min
    let minStability: Double
    let finalStability: Double
    let worstThermalState: Int // 0 = Nominal, 1 = Fair, 2 = Serious, 3 = Critical
    let stamps: [DeviceHammerStamp]
    let deviceModel: String
    let deviceManufacturer: String
    let osVersion: String
    let sessionId: Int
    let encryptionKey: String
}

class PendingResultStore: ObservableObject {
    static let shared = PendingResultStore()
    
    @Published var results: [PendingTestResult] = []
    
    private let fileURL: URL = {
        let paths = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
        return paths[0].appendingPathComponent("pending_results.json")
    }()
    
    private init() {
        loadResults()
    }
    
    func saveResult(_ result: PendingTestResult) {
        results.append(result)
        persist()
    }
    
    func deleteResult(id: UUID) {
        results.removeAll { $0.id == id }
        persist()
    }
    
    private func loadResults() {
        guard FileManager.default.fileExists(atPath: fileURL.path) else { return }
        do {
            let data = try Data(contentsOf: fileURL)
            let decoder = JSONDecoder()
            self.results = try decoder.decode([PendingTestResult].self, from: data)
        } catch {
            print("Failed to load pending results: \(error)")
        }
    }
    
    private func persist() {
        do {
            let encoder = JSONEncoder()
            let data = try encoder.encode(results)
            try data.write(to: fileURL, options: [.atomic, .completeFileProtection])
        } catch {
            print("Failed to persist pending results: \(error)")
        }
    }
}

File: ThermoHammerAndroid/app/src/main/java/com/example/thermohammer/data/PendingResultStore.kt
package com.example.thermohammer.data

import android.content.Context
import com.example.thermohammer.network.DeviceHammerStamp
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.io.File

data class PendingTestResult(
    val id: String, // Unique local identifier
    val timestamp: Long,
    val durationSeconds: Int,
    val testDurationType: Int, // 0=5min, 1=15min, 2=30min
    val minStability: Float,
    val finalStability: Float,
    val worstThermalState: Int, // 0=Nominal, 1=Fair, 2=Serious, 3=Critical
    val stamps: List<DeviceHammerStamp>,
    val deviceModel: String,
    val deviceManufacturer: String,
    val osVersion: String,
    val sessionId: Int,
    val encryptionKey: String
)

class PendingResultStore(context: Context) {
    private val file = File(context.filesDir, "pending_results.json")
    private val gson = Gson()

    fun saveResult(result: PendingTestResult) {
        val list = getAllResults().toMutableList()
        list.add(result)
        file.writeText(gson.toJson(list))
    }

    fun getAllResults(): List<PendingTestResult> {
        if (!file.exists()) return emptyList()
        return try {
            val type = object : TypeToken<List<PendingTestResult>>() {}.type
            gson.fromJson(file.readText(), type) ?: emptyList()
        } catch (e: Exception) {
            emptyList()
        }
    }

    fun deleteResult(id: String) {
        val list = getAllResults().filter { it.id != id }
        file.writeText(gson.toJson(list))
    }
}
